signing/mcl/singlesig/bls_test.go

```diff
@@ -244,3 +248,84 @@ func TestBLSSigner_IsInterfaceNil(t *testing.T) {
 
 	require.False(t, check.IfNil(llSig))
 }
+
+type TestVector struct {
+	TestName     string `json:"testName"`
+	SecretKeyHex string `json:"secretKeyHex"`
+	PublicKeyHex string `json:"publicKeyHex"`
+	Message      string `json:"message"`
+	Signature    string `json:"signature"`
+	Error        string `json:"error"`
+}
+type TestVectors struct {
+	TestVectors []TestVector `json:"testVectorsVerify"`
+}
+
+func TestBLSSigner_TestVectorsSign(t *testing.T) {
+	t.Parallel()
+
+	jsonFile, err := os.Open("./SingleSignTestVectorsSign.json")
+	require.Nil(t, err)
+	defer jsonFile.Close()
+	var testVar TestVectors
+	jsonDec := json.NewDecoder(jsonFile)
+	err = jsonDec.Decode(&testVar)
+	require.Nil(t, err)
+
+	signer := singlesig.NewBlsSigner()
+	suite := mcl.NewSuiteBLS12()
+	kg := signing.NewKeyGenerator(suite)
+
+	for _, testVector := range testVar.TestVectors {
+		if len(testVector.TestName) == 0 {
+			continue
+		}
+		t.Run(testVector.TestName, func(t *testing.T) {
+			secretKeyBytes, err := hex.DecodeString(testVector.SecretKeyHex)
+			require.Nil(t, err)
+			sk, err := kg.PrivateKeyFromByteArray(secretKeyBytes)
+			require.Nil(t, err)
+			signatureBytes, err := signer.Sign(sk, []byte(testVector.Message))
+			signature := hex.EncodeToString(signatureBytes)
+			errorString := ""
+			if err != nil {
+				errorString = err.Error()
+			}
+			require.Equal(t, signature, testVector.Signature)
+			require.Equal(t, errorString, testVector.Error)
```

Comment:
```suggestion
			secretKeyBytes, err := hex.DecodeString(testVector.SecretKeyHex)
			require.Nil(t, err)
			
			sk, err := kg.PrivateKeyFromByteArray(secretKeyBytes)
			require.Nil(t, err)
			
			signatureBytes, err := signer.Sign(sk, []byte(testVector.Message))
			errorString := ""
			if err != nil {
				errorString = err.Error()
			}
			signature := hex.EncodeToString(signatureBytes)
			
			require.Equal(t, testVector.Signature, signature)
			require.Equal(t, testVector.Error, errorString)
```

Reply:
added some spacing + treated error first + updated the requires to have the expected value first